src/mongocxx/test/hint.cpp

```diff
@@ -52,6 +52,22 @@ TEST_CASE("Hint", "[hint]") {
             auto index_doc = builder::stream::document{} << "a" << 1 << builder::stream::finalize;
             REQUIRE(index_hint != index_doc);
         }
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
Filed https://jira.mongodb.org/browse/CXX-1056 for suppressing deprecation warnings in testing.